Sources_v3/Database/DatabaseSession.swift

```diff
@@ -133,6 +133,10 @@ extension DatabaseSession {
             try saveCurrentUserUnreadCount(count: unreadCount)
         }
         
+        if let currentUser = currentUser() {
+            currentUser.lastReceivedEventDate = payload.createdAt ?? Date()
+        }
+        
```

Comment:
Let's not save anything if `createdAt` is `nil`. As mentioned in Slack, it's better to have duplicate events rather than miss some.